pants_plugins/sendwave/pants_docker/python_requirement.py

```diff
@@ -93,12 +90,56 @@ async def get_requirements(
     constraint_arg = ""
     if setup.requirement_constraints:
         constraint_arg = f"--constraint {setup.requirement_constraints}"
+    return PipInstallArgs(
+        links_args=links_args,
+        index_args=index_args,
+        constraint_arg=constraint_arg,
+    )
+
+
+@dataclass(frozen=True)
+class PythonRequirementsFS:
+    requirements: Tuple[PythonRequirementsField]
+
+
+@rule
+async def get_requirements(
+    field_set: PythonRequirementsFS, setup: PythonSetup, repos: PythonRepos
+) -> DockerComponent:
+    install_args = _get_install_args(setup, repos)
+    commands = (
+        f"RUN python -m pip install {install_args.index_args} "
+        + f"{install_args.links_args} {install_args.constraint_arg} "
+        + " ".join(str(lib) for lib in field_set.requirements)
+        + "\n"
+    )
+    return DockerComponent(
+        commands=commands,
+        sources=None,
+    )
 
+
+@dataclass(frozen=True)
+class MultilinePythonRequirementsFS:
+    requirements: Tuple[PythonRequirementsField]
```

Comment:
since this is the same data as the single line pip install - do you think it would be cleaner to have one dataclass & one rule to get a docker component for the 3rd party python requirements altogether. And switch on the subsystem option in that rule?

(in fact it could also include the commands to create the virtual environment too? But that might be too big of a refactor)

Reply:
Yeah, I thought about going in this direction at first, but in the end I assumed it might be useful to configure this differently for different targets. Still, I think you're right that it would make the rules much simpler to use a subsystem option, and for our own purposes we don't really care about configuring it differently for individual targets. So I guess I'll move it to a subsystem option then!